### PR TITLE
Support for DGF muscle tendon compliance in CMC and forward simulations 

### PR DESCRIPTION
**Pull request by nickbianco**
Fixes issue #2714

### Brief summary of changes
- Incorporated all changes implemented in branch `feature_dgf_muscle` (already reviewed by @carmichaelong via #2793)
- `DeGrooteFregly2016Muscle` now properly computes muscle-tendon equilibrium via `calcEquilibriumResidual()`, which is used by `computeInitialFiberEquilibrium()`. It previously always produced a zero residual since it relied on advancing the SimTK::Stage, but now it is independent of Stage.
- Created new test cases to test that `DeGrooteFregly2016Muscle` with tendon compliance enabled can produce the correct trajectory and/or muscle activity when using forward integration or CMC
     - Includes tests in `testDeGrooteFregly2016Muscle.cpp` and `testMuscles.cpp`
- `MocoInitialForceEquilibriumGoal` is now `MocoInitialForceEquilibriumDGFGoal` with changes that are specific to the new changes in how the DGF muscle computes muscle-tendon equilbrium. 

### Looking for feedback on...
In `CMCTool`, I wrapped a call to `Muscle::computeEquilibrium()` with an existing bool property since it was causing the CMC tests I added to fail when disabling activation dynamics. @chrisdembia, @carmichaelong, and I discussed this during the dev meeting and were not sure if CMC generally supports problems with activation dynamics disabled anyway, so we may choose to punt on these tests for now.

### CHANGELOG.md (choose one)

- no need to update because...still part of Moco feature branch.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2859)

### CODE REVIEW

**Comment by nickbianco** [comment]
After quickly going through the CMC tests, it doesn't look like we have tests that ignore activation dynamics.

File: OpenSim/Actuators/DeGrooteFregly2016Muscle.cpp [hunk 1]
```diff
@@ -279,9 +299,8 @@ void DeGrooteFregly2016Muscle::calcMuscleLengthInfoHelper(
 void DeGrooteFregly2016Muscle::calcFiberVelocityInfoHelper(
         const SimTK::Real& muscleTendonVelocity, const SimTK::Real& activation,
         const bool& ignoreTendonCompliance,
-        const bool& isTendonDynamicsExplicit,
-        const MuscleLengthInfo& mli, FiberVelocityInfo& fvi,
-        const SimTK::Real& normTendonForce = SimTK::NaN,
+        const bool& isTendonDynamicsExplicit, const MuscleLengthInfo& mli,
```
Comment (adamkewley) [inline OpenSim/Actuators/DeGrooteFregly2016Muscle.cpp:302]: Neither a review, nor blocking, nor do you have to answer, but why is it that devs default to using `const T&` for primitive value types like `bool` or `SimTK::Real`?

Comment (chrisdembia) [reply OpenSim/Actuators/DeGrooteFregly2016Muscle.cpp:302]: No good reason. We probably should pass such primitive types by value.

**Comment by chrisdembia** [comment]
DGF + CMC :)

**Pull request merged**